src/lib/routing-table.js

```diff
@@ -35,6 +35,28 @@ class RoutingTable {
     return false
   }
 
+  /**
+   * Compute a curve quote's `appliesToPrefix`, which is the shortest prefix
+   * that uniquely matches the target.
+   *
+   * @param {IlpAddress} routePrefix
+   * @param {IlpAddress} destinationAccount
+   * @returns {IlpAddress}
+   */
+  getAppliesToPrefix (routePrefix, destinationAccount) {
+    // Use `routePrefix` as the initial `appliesToPrefix`, and extend it as
+    // needed if it is too general.
+    let appliesToPrefix = routePrefix
+    this.destinations.each((routes, targetPrefix) => {
+      if (targetPrefix === routePrefix) return
+      if (targetPrefix.startsWith(appliesToPrefix)) {
+        appliesToPrefix = destinationAccount.slice(
+          0, getPrefixLength(targetPrefix, appliesToPrefix) + 1)
```

Comment:
if you already know that targetPrefix starts with appliesToPrefix (see line 52), then `getPrefixLength(targetPrefix, appliesToPrefix)` will always return `appliesToPrefix.length`, right?